src/stpipe/step.py

```diff
@@ -56,6 +56,10 @@ class Step:
     # name.  Must be globally unique!
     class_alias = None
 
+    # Default formatting of output filenames unless overridden, typically with
+    # a format string provided along with output_use_model
+    name_format = None
```

Comment:
Maybe this comment can be closer to Jonathan's description:

"String defining the format of the output name which defines how **components are inserted into the output file name. If None, use the default formatting, which is to simply append Step.suffix to the name. However, if False, use basepath as its own format string."